### PR TITLE
Remove type flags information from generated hash

### LINKED ISSUE #30816: "No such type" error in RC5
I tried switching everything to a single package.  I put it in this branch

https://github.com/ballerina-platform/nballerina/tree/james-single-package

It compiles OK with (bal build --skip-tests), but when I run e.g. using

```
bal run -- --showTypes tests\data\basic.bal
```

I get:

```
 No such type: Package: ballerina/lang$0046__internal/0_1_0/$_init, TypeName: $type$$anonType$_12, Shape: record {| int value; |}
```

Not sure if this is the same as #30814.

### PR DESCRIPTION
**Pull request by grainier**
Fixes #30816

## Approach
n/a

## Samples
n/a

## Remarks
n/a

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/jballerina-unit-test/src/test/resources/test-src/expressions/binaryoperations/integer_range_operators.bal [hunk 1]
```diff
@@ -1,3 +1,5 @@
+public type Label int; // If the TypeHashVisitor takes in type.flags of `Lable` type, int range tests will fail.
+
```
Comment (MaryamZi) [inline tests/jballerina-unit-test/src/test/resources/test-src/expressions/binaryoperations/integer_range_operators.bal:2]: Shall we move this out to a separate file? And run and assert an iteration maybe?
```suggestion
public type Label int; // If the TypeHashVisitor takes in type.flags of `Label` type, int range tests will fail.

```

**Review by MaryamZi — APPROVED** [review]

File: tests/jballerina-unit-test/src/test/resources/test-src/expressions/binaryoperations/integer-range-operators-runtime.bal [hunk 2]
```diff
@@ -0,0 +1,11 @@
+public type Label int; // If the TypeHashVisitor takes in type.flags of `Label` type, there will be runtime faliures.
```
Comment (MaryamZi) [inline tests/jballerina-unit-test/src/test/resources/test-src/expressions/binaryoperations/integer-range-operators-runtime.bal:1]: Missing the licence header.

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/expressions/binaryoperations/IntegerRangeOperatorTest.java [hunk 3]
```diff
@@ -170,6 +173,23 @@ public void testHalfOpenIntRangeAsArray() {
...
+        BValue[] returns = BRunUtil.invoke(runtimeResult, "testClosedIntRange", args);
+        Assert.assertEquals(returns.length, 1);
+
+        BValueArray returnArray = (BValueArray) returns[0];
+        int expectedSize = endValue - startValue + 1;
+        Assert.assertEquals(returnArray.size(), expectedSize, "Incorrect number of values returned for Closed "
+                + "Integer Range " + startValue + " ... " + endValue);
+        for (int i = 0; i < expectedSize; i++) {
+            Assert.assertEquals(returnArray.getInt(i), startValue + i, "Incorrect value found at index: " + i);
+        }
```
Comment (MaryamZi) [inline tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/expressions/binaryoperations/IntegerRangeOperatorTest.java:190]: We can write this in b7a itself.

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/expressions/binaryoperations/IntegerRangeOperatorTest.java [hunk 4]
```diff
@@ -34,11 +34,14 @@
...
 
     private CompileResult result;
+    private CompileResult runtimeResult;
     private CompileResult negativeResult;
 
     @BeforeClass
     public void setup() {
         result = BCompileUtil.compile("test-src/expressions/binaryoperations/integer_range_operators.bal");
+        runtimeResult = BCompileUtil.
+                compile("test-src/expressions/binaryoperations/integer-range-operators-runtime.bal");
```
Comment (MaryamZi) [inline tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/expressions/binaryoperations/IntegerRangeOperatorTest.java:44]: We can probably have a separate package for the type hash related tests, right?

How do we test the TypeHashVisitor btw? If not unit tests, I think we need to have at least tests with multiple anon types that may or may not result in the same hash. Shall we create an issue for this?

**Review by MaryamZi — APPROVED** [review]

File: tests/jballerina-unit-test/src/test/resources/test-src/expressions/binaryoperations/integer-range-operators-runtime.bal (hunk 2 above)
Comment (grainier) [reply tests/jballerina-unit-test/src/test/resources/test-src/expressions/binaryoperations/integer-range-operators-runtime.bal:17]: 83a6739

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/expressions/binaryoperations/IntegerRangeOperatorTest.java (hunk 3 above)
Comment (grainier) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/expressions/binaryoperations/IntegerRangeOperatorTest.java]: 47c3db3

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/expressions/binaryoperations/IntegerRangeOperatorTest.java (hunk 4 above)
Comment (grainier) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/expressions/binaryoperations/IntegerRangeOperatorTest.java:44]: Created https://github.com/ballerina-platform/ballerina-lang/issues/30849

**Review by MaryamZi — APPROVED** [review]

**Pull request merged**